library/src/plugins/official/backend/actions/sse.ts

```diff
@@ -138,15 +146,47 @@ export const SSE: ActionPlugin = {
       }
 
       const urlInstance = new URL(url, window.location.origin)
-      const json = signals.JSON(false, !includeLocal)
-      if (method === 'GET') {
-        const queryParams = new URLSearchParams(urlInstance.search)
-        queryParams.set(DATASTAR, json)
-        urlInstance.search = queryParams.toString()
+      const queryParams = new URLSearchParams(urlInstance.search)
+      
+      if (contentType === 'json') {
+        const json = signals.JSON(false, !includeLocal)
+        if (method === 'GET') {
+          queryParams.set(DATASTAR, json)
+        } else {
+          req.body = json
+        }
+      } else if (contentType === 'form') {
+        const formEl = selector ? document.querySelector(selector) : el.closest('form');
+        if (formEl === null) {
+          if (selector) {
+            throw dsErr('SseFormNotFound', { selector })
+          } else {
+            throw dsErr('SseClosestFormNotFound')
+          }
+        }
+        const preventSubmit = (evt: Event) => evt.preventDefault()
+        formEl.addEventListener('submit', preventSubmit)
```

Comment:
If somebody calls `sse` from a data-on-submit, I dont think this will prevent it's default behaviour. This would probably come as a surprise to anybody trying to use data-on-submit.

Reply:
You’re right. Is there anyway around this, besides documenting that `data-on-submit__prevent` must be used?

Reply:
maybe always prevent if it onSubmit?  If you are using Datastar, you are always preventing default form stuff

Reply:
Ok, `data-on-submit` attributes now have `preventDefault()` applied to the event by default.